src/Sarif/Baseline/ResultMatching/SarifLogResultMatcher.cs

```diff
@@ -190,37 +190,53 @@ private List<ExtractedResult> ExtractResultsFromRuns(IEnumerable<Run> sarifRuns)
             {
                 throw new ArgumentException(nameof(currentRuns));
             }
-            
+
             // Results should all be from the same tool, so we'll pull the log from the first run.
-            Tool tool = currentRuns.First().Tool.DeepClone();
+            Run firstRun = currentRuns.First();
+            Tool tool = firstRun.Tool.DeepClone();
+
+            // Only include the rules corresponding to matched results.
+            tool.Driver.Rules = null;
 
             var run = new Run
             {
-                Tool = tool,
-                AutomationDetails = currentRuns.First().AutomationDetails,
+                Tool = tool
             };
 
+            // If there was only one run, we can fill in more information because we don't need to
+            // worry about it being different from run to run.
+            if (currentRuns.Count() == 1)
+            {
+                run.AutomationDetails = firstRun.AutomationDetails;
```

Comment:
>run [](start = 16, length = 3)

help me remember, why in this case don't we simply use 'firstRun' in its entirety?

Reply:
I'm just not 100% sure that everything makes sense to copy. I was confident of these and it was a large superset of what Dan asked for.

---
In reply to: [328779252](https://github.com/microsoft/sarif-sdk/pull/1686#discussion_r328779252) [](ancestors = 328779252)

Reply:
Eventually we should have a precise understanding of this. i.e., it is a behavior that is worth consciously specifying. If this area breaks again, we really should consider doing that. This is a good change that unblocks a colleague. Currently, however this specific behavior (and its corresponding unit test) are on of those 'implementation has a de facto/poorly explained behavior and our test simply ensures there is no change in this unexplained behavior. Absolutely better than no testing, obviously.

---
In reply to: [328779874](https://github.com/microsoft/sarif-sdk/pull/1686#discussion_r328779874) [](ancestors = 328779874,328779252)

Reply:
Agreed.

Note that the other part of this change (the rules array and the references to it from the results array) _is_ based on a precise understanding. The existing code was just wrong.

---
In reply to: [328783776](https://github.com/microsoft/sarif-sdk/pull/1686#discussion_r328783776) [](ancestors = 328783776,328779874,328779252)